src/python/pants/backend/python/lint/pylint/rules.py

```diff
@@ -0,0 +1,115 @@
+  target: TargetAdaptor
+
+
+def generate_args(file_names: Tuple[str, ...], pylint: Pylint) -> Tuple[str, ...]:
+  args = []
+  if pylint.options.config is not None:
+    args.append(f"--config={pylint.options.config}")
+  args.extend(pylint.options.args)
+  args.extend(file_names)
+  return tuple(args)
+
+
+@rule(name="Lint using Pylint")
+async def lint(
+  wrapped_target: PylintTarget,
+  pylint: Pylint,
+  python_setup: PythonSetup,
+  subprocess_encoding_environment: SubprocessEncodingEnvironment,
+) -> LintResult:
+  if pylint.options.skip:
+    return LintResult.noop()
+
+  target = wrapped_target.target
+
+  hydrated_target = await Get[HydratedTarget](Address, target.address)
+  dependencies = await MultiGet(
+    Get[HydratedTarget](Address, dependency)
+    for dependency in hydrated_target.dependencies
+  )
+  sources_digest = await Get[ChrootedPythonSources](HydratedTargets([hydrated_target, *dependencies]))
+
+  source_root_stripped_target_sources = await Get[SourceRootStrippedSources](
+    Address, target.address
+  )
+  file_names = source_root_stripped_target_sources.snapshot.files
+
+  # NB: Pylint output depends upon which Python interpreter version it's run with. We ensure that
+  # each target runs with its own interpreter constraints. See
+  # http://pylint.pycqa.org/en/latest/faq.html#what-versions-of-python-is-pylint-supporting.
+  interpreter_constraints = PexInterpreterConstraints.create_from_adaptors(
+    adaptors=[target] if isinstance(target, PythonTargetAdaptor) else [],
+    python_setup=python_setup
+  )
+
+  config_path: Optional[str] = pylint.options.config
+  config_snapshot = await Get[Snapshot](
+    PathGlobs(
+      globs=tuple([config_path] if config_path else []),
+      glob_match_error_behavior=GlobMatchErrorBehavior.error,
+      description_of_origin="the option `--pylint-config`",
+    )
+  )
+  requirements_pex = await Get[Pex](
+    CreatePex(
+      output_filename="pylint.pex",
+      requirements=PexRequirements(requirements=tuple(pylint.get_requirement_specs())),
+      interpreter_constraints=interpreter_constraints,
+      entry_point=pylint.get_entry_point(),
+    )
+  )
+
+  merged_input_files = await Get[Digest](
+    DirectoriesToMerge(
+      directories=(
+        requirements_pex.directory_digest,
+        config_snapshot.directory_digest,
+        sources_digest.digest,
+      )
+    ),
+  )
+  request = requirements_pex.create_execute_request(
+    python_setup=python_setup,
+    subprocess_encoding_environment=subprocess_encoding_environment,
+    pex_path=f'./pylint.pex',
+    pex_args=generate_args(file_names, pylint),
+    input_files=merged_input_files,
+    description=f'Run Pylint for {target.address.reference()}',
+  )
+  result = await Get[FallibleExecuteProcessResult](ExecuteProcessRequest, request)
+  return LintResult.from_fallible_execute_process_result(result)
+
+
+def rules():
+  return [lint, subsystem_rule(Pylint), UnionRule(PythonLintTarget, PylintTarget)]
```

Comment:
We recently made a change so that you no longer need to register `pants.backend.python` for `pants.backend.python.lint.*` to work. See https://github.com/pantsbuild/pants/pull/9059

Here, you'll want to register any rule that is needed to get this working. You can copy them from your tests' `rules()` function (and then remove it from the tests). See the PR for some examples of what this looks like.